### ISSUE TITLE
YeeLight's FW updated, cannot get maximal brightness

### ISSUE THREAD

**Issue by biogerm** [vvpossible/homebridge_yeelight#63]
Hi,

YeeLight's FW was updated last week and I can't get to the maximal brightness anymore with home kit app. Anyone else seeing this issue?

BR
An Qin

**Comment by biogerm** [comment]
seems only affect in color temperature mode. My guess is yeelight changed the way to interpret color as mentioned in FW update log. then it treats color temp mode message from home kit as color mode message.

**Comment by iChUdY** [comment]
Yes. This plugin send only HSB (Color) messages. Color temp is different CT messages.. need rewrite this plugin for work with Color Temperature.

**Comment by happytobeunhinged** [comment]
This has to be the most frustrating issue I have. Affects all lights in the house. Any homekit control means the brightness isn't in "sun mode". Work around is to set the brightness to 100% in yeelight app and save as default state then power cycle the light manually to get it bright.

**Comment by vvpossible** [comment]
Yeelight bulbs treat color temperature and color differently (under CT mode, Yeelight lamps will mix lots of WHITE to increase the brightness in order to make the device more usable), however, Homekit uses HSV to change color as well as CT. There is a color temperature characteristic defined in Homekit Spec, but it also explicitly specify that this characteristic can't co-exist with Hue/Saturation.  
To fix this, Yeelight firmware needs to implement HSV to CT conversion, it's not an easy task.

**Comment by iChUdY** [comment]
@vvpossible no need to change yeelight fw. Homekit two way control. From Color, from Temperature. Viz. https://github.com/vieira/homebridge-yeelight-wifi/pull/25

This issue possible fix from homebridge plugin code.

**Comment by J7mbo** [comment]
#25 was closed recently.
